File: 1996-The Number of Weak Characters in the Game.py
class Solution(object):
    def numberOfWeakCharacters(self, properties):
        """
        :type properties: List[List[int]]
        :rtype: int
        """
        properties.sort(key = lambda x: (-x[0], x[1]))
        
        maxDefense = 0
        res = 0
        
        for _, defense in properties:
            if defense < maxDefense:
                res += 1
            if defense > maxDefense:
                maxDefense = defense
                
        return res

File: JavaScript/1996-The Number of Weak Characters in the Game.js
/**
 * @param {number[][]} properties
 * @return {number}
 */
var numberOfWeakCharacters = function(properties) {
    properties.sort(function(a,b){
        return a[0] === b[0] ? (a[1]-b[1]) : (b[0] - a[0]);
    })
    
    let maxDefense = 0, res = 0;
    
    for (let [attack, defense] of properties){
        if (defense < maxDefense) res++;
        if (defense > maxDefense) maxDefense = defense;
    }
    return res;
};